docs/core/tools/dotnet-build.md

```diff
@@ -77,9 +75,15 @@ Running `dotnet build` is equivalent to running `dotnet msbuild -restore`; howev
 
 ## Arguments
 
-`PROJECT | SOLUTION`
+`PROJECT | SOLUTION | FILE`
+
+The project or solution or C# (file-based program) file to publish. If a file isn't specified, MSBuild searches the current directory for a project or solution.
+
+- `PROJECT` is the path and filename of a C#, F#, or Visual Basic project file, or the path to a directory that contains a C#, F#, or Visual Basic project file.
 
-The project or solution file to build. If a project or solution file isn't specified, MSBuild searches the current working directory for a file that has a file extension that ends in either *proj* or *sln* and uses that file.
+- `SOLUTION` is the path and filename of a solution file (*.sln* or *.slnx* extension), or the path to a directory that contains a solution file.
+
+- `FILE` is an argument added in .NET 10. The path and filename of a file-based program. File-based programs are contained within a single `*.cs` file that are built and run without a corresponding project (`*.csproj`) file. For more information, see [Build file-based C# programs](/dotnet/csharp/fundamentals/tutorials/file-based-programs).
```

Comment:
I don't know if it's needed here, but the `*.cs` extension is a convention, but not required. The reason is `#!` support on unix systems. For example, if you have `my-utility.cs`, you could name it `my-utility` and add the `#!/usr/local/share/dotnet/dotnet run` as the first line of the file.